native/cocos/renderer/gfx-vulkan/VKGPUObjects.h

```diff
@@ -1160,6 +1142,47 @@ class CCVKGPURecycleBin final {
     size_t _count = 0U;
 };
 
+class CCVKGPUInputAssemblerHub {
+public:
+    explicit CCVKGPUInputAssemblerHub(CCVKGPUDevice *device)
+    : _gpuDevice(device) {
+    }
+
+    ~CCVKGPUInputAssemblerHub() = default;
+
+    void connect(CCVKGPUInputAssembler *ia, const CCVKGPUBufferView *buffer) {
+        _ias[buffer].insert(ia);
+    }
+
+    void update(CCVKGPUBufferView* oldBuffer, CCVKGPUBufferView* newBuffer) {
+        auto iter = _ias.find(oldBuffer);
+        if (iter != _ias.end()) {
+            for (const auto &ia : iter->second) {
+                ia->update(oldBuffer, newBuffer);
+                _ias[newBuffer].insert(ia);
+            }
+            _ias.erase(iter);
+        }
+    }
+
+    void disengage(const CCVKGPUBufferView *buffer) {
+        auto iter = _ias.find(buffer);
+        if (iter != _ias.end()) {
+            _ias.erase(iter);
+        }
+    }
+
+    void disengage(CCVKGPUInputAssembler *set, const CCVKGPUBufferView *buffer) {
+        auto iter = _ias.find(buffer);
+        if (iter == _ias.end()) return;
+        iter->second.erase(set);
```

Comment:
The logic of two disengage functions are better to have the coding style. Currently they are a little different:
```
if (iter != _ias.end()) {
            _ias.erase(iter);
        }

if (iter == _ias.end()) return;
        iter->second.erase(set);
```